### PR TITLE
[1.13.x] Jenkinsfile.prod.nightly optaplanner-quickstarts scmRevision updated

### PR DESCRIPTION
**Pull request by Ginxo**

### CODE REVIEW

File: .ci/jenkins/Jenkinsfile.prod.nightly [hunk 1]
```diff
@@ -64,7 +64,7 @@ pipeline {
                             def SETTINGS_XML_ID = RHBA_RELEASE_VERSION ? 'kogito-nightly-version' : 'kogito-nightly-main'
                             def projectCollection = ['kogito-runtimes', 'optaplanner', 'kogito-examples', 'optaplanner-quickstarts', 'optaweb-vehicle-routing', 'optaweb-employee-rostering', 'jboss-integration/kogito-rhba']
                             def projectVariableMap = [:]
-                            def additionalVariables = ['optaplannerProductVersion': env.OPTAPLANNER_PRODUCT_VERSION, 'optaplanner-scmRevision': getOptaPlannerBranch(), 'optaplanner-quickstarts-scmRevision': getOptaPlannerBranch('stable'), 'rhbaProductVersion': env.RHBA_LASTEST_PRODUCT_VERSION]
+                            def additionalVariables = ['optaplannerProductVersion': env.OPTAPLANNER_PRODUCT_VERSION, 'optaplanner-scmRevision': getOptaPlannerBranch(), 'optaplanner-quickstarts-scmRevision': getOptaPlannerBranch(), 'rhbaProductVersion': env.RHBA_LASTEST_PRODUCT_VERSION]
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.prod.nightly]: I guess we should backport some changes which were done on main.
See https://github.com/kiegroup/kogito-pipelines/blob/main/.ci/jenkins/Jenkinsfile.prod.nightly#L60

But not the drools part ;)

Comment (Ginxo) [reply .ci/jenkins/Jenkinsfile.prod.nightly]: @radtriste done!! thanks!

File: .ci/jenkins/Jenkinsfile.prod.nightly [hunk 2]
```diff
@@ -186,16 +193,25 @@ def treatGitHashes(String gitInformationHashes) {
...
+}
 
+String getOptaPlannerQuickstartsBranch() {
+    def branch = env.CHANGE_BRANCH ?: BRANCH_NAME
+    return branch == 'main' ? 'development' : calculateKieRepoBranch(branch)
+}
+
+String calculateKieRepoBranch(String branch) {
+    /* The Drools/OptaPlanner major version is shifted by 7 from the Kogito major version:
+    Kogito 1.x.y -> Drools 8.x.y.
```
Comment (mareknovotny) [inline .ci/jenkins/Jenkinsfile.prod.nightly]: Drools should be the same as branched 7.x -> I am guessing at beginning of next sprint it will be 7.67.x

Comment (Ginxo) [reply .ci/jenkins/Jenkinsfile.prod.nightly]: Drools is not even part of the process, I should probably remove that comment

Comment (Ginxo) [reply .ci/jenkins/Jenkinsfile.prod.nightly]: done

**Review by radtriste — APPROVED** [review]

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**